meilisearch/src/routes/indexes/mod.rs

```diff
@@ -120,7 +120,8 @@ pub async fn create_index(
 ) -> Result<HttpResponse, ResponseError> {
     let IndexCreateRequest { primary_key, uid } = body.into_inner();
 
-    let allow_index_creation = index_scheduler.filters().search_rules.is_index_authorized(&uid);
+    // FIXME: allow_index_creation?
+    let allow_index_creation = index_scheduler.filters().is_index_authorized(&uid);
```

Comment:
What is this FIXME?

Reply:
I was surprised by the code here: we determine `allow_index_creation` from `is_index_authorized`. 

I would expect it be determined from `filters().allow_index_creation()`

Reply:
Note that although the PR changes the variable to use the `filters.is_index_authorized()` method, the original code would use `filters.search_rules.is_index_authorized()` which explains why it is surprising

Reply:
Update: following internal discussion, I updated the `AuthFilter::allow_index_creation` to take an indexUid and check both for access with this uid and creation permission.

I updated all the call sites and so removed the FIXME

Reply:
noice 👌